tests/test_pgp.ml

```diff
@@ -0,0 +1,254 @@
+    in
+    let u =
+      Z.of_string
+        "0xaf96592890d66e7c762d11774d700877a3dbc1406a843547fe63f36715cdc11e0f0e8174577ca807dde961358ee44f59170a8ebd1eedc98f0c0b22125c111e6d72ebbaa2a2a477d492840ccbd0ca7f77915fa13b891da483e689671a7321176cb851605ef2970fbe4c8d3c33fb5f48c52ff9e0c7803e5bc6db000dcbf82b77772fc09a6405d1166b6cd8ac1f1b97d142d4730d31deb02f5bed6b12625ab3bc7d46150e2ee5ab432d02e2ccbec991cb4b71b8d43c7b9ef90807c909b302fd11da"
+    in
+
+    let private_key = Rsa.Private.{d; p; q; u} in
+    let packet =
+      Packet.Secret_key.
+        { public_key = expected_public2
+        ; s2k = None
+        ; initial_vector = None
+        ; private_key = Some (Rsa private_key)
+        ; checksum = Some "e90c"
+        ; hash = None }
+    in
+    Packet.Body.Secret_key packet
+
+  let test_rsa_pub1 = test_val ~decode public_packet "rsa_public.pgp"
+
+  let treat_secret1 (res : Packet.t list) =
+    let res1 = List.hd res in
+    match res1.packet with
+    | Secret_key packet ->
+      let public_packet = packet.public_key in
+      let public_key =
+        match public_packet.public_key with
+        | Rsa key -> key
+        | Dsa _ -> raise (File "Rsa_private.pgp is not the right file")
+        | _ -> raise (File "Rsa_private.pgp is not the right file")
+      in
+      public_key
+    | _ -> raise (File "Rsa_private.pgp is not the right file.")
+
+  let public_key2 =
+    let n =
+      Z.of_string
+        "0xc64c9e5d63cf62479608c22ba50cf2cbda2497e2cd92bdbb536ab4d55f1befaf262ae89a739698e538c688bce698aa4258bff01464f714ce98a2b1bacf92871f6e2b040990924f2c6634804072a990c6328ccd580e01f54b013aa0e08a388a78cd20ece56ca865079c26e77875c3b1f42d6bf015bb95da4606bf3fb05608ec17f823373dd73b75d77d1a34568797a690cbd21c6c78dd99d4f8cbea71e0327e8ca544619c244d1e9fa1a624209a7986b8663a1b1b0044899120d0daffddcf06df12159f8962c433a3b3ae7d84c5630ad290d97921b21a318a80d6a3b702d51cb2ba23958141a1702096cf891794f929e8396645f697e4f928a8105102f2ffed7b2ea920ffea992e74bfaeef564a6a06a7a61a0a379446367de345e9865205d095be574bb3bbc47721f71d4adaa79e2e4dcc781c94f90c64f32720a61b8dda1e6cf7093a2144276f772844242444cb06e7cf42a7745baaf4eb40b7975cc89aa32ff1d34fa8cadfc5a6aae59fff707c0d225e943ad479585af4bed5b71e78e183f3"
+    in
+    let e = Z.of_string "0x10001" in
+    Rsa.Public.{n; e}
+
+  let test_rsa_public1 = test_val ~decode public_packet "rsa_public.pgp"
+
+  let test_rsa_public2 ctxt =
+    let res = decode (fixture "rsa_private.pgp") in
+    let res_public = treat_secret1 res in
+    assert_equal ~ctxt res_public public_key2
+
+  let test_rsa_private = test_val ~decode secret_packet "rsa_private.pgp"
```

Comment:
I'm a bit confused by these two tests (`test_rsa_public1` and `test_rsa_private`).  Isn't there a way to check that the right key is returned in just one "assert"?  Or am I missing something?